### PR TITLE
feat: Support custom status code for openapi exceptions

### PR DESCRIPTION
**Pull request by abdulhaq-e**
### Pull Request Checklist

[//]: # "Please review the [Litestar contribution guidelines](https://github.com/litestar-org/litestar/blob/main/CONTRIBUTING.rst) for this repository."

- [x] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

### Description

If the exception has a `detail` field, that will be used as the description and example.

[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

### Close Issue(s)

https://github.com/litestar-org/litestar/issues/1642

[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

### CODE REVIEW

**Comment by abdulhaq-e** [comment]
If approved, I'll update docs as well.

**Comment by abdulhaq-e** [comment]
I think a better approach would be to create something like `CustomHttpException` protocol which forces implementing `detail` and `description` and then all custom exceptions for OpenAPI has to implement that protocol.

**Comment by Goldziher** [comment]
The approach here is ok, please bring this PR home

**Comment by Goldziher** [comment]
@abdulhaq-e ?

**Comment by abdulhaq-e** [comment]
> @abdulhaq-e ?

I'll rebase and add docs. Then it's ready. Will do it today.

**Review by Goldziher — COMMENTED** [review]
Code is good 👍.

File: litestar/_openapi/responses.py [hunk 1]
```diff
@@ -184,28 +185,45 @@ def create_error_responses(exceptions: list[type[HTTPException]]) -> Iterator[tu
...
+        exceptions_schemas = []
+        group_description: str = ""
+        for exc in exception_group:
+            example_detail = ""
+            if hasattr(exc, "detail") and exc.detail:
+                group_description = exc.detail
+                example_detail = exc.detail
+
+            if not example_detail:
+                with contextlib.suppress(Exception):
```
Comment (Goldziher) [inline litestar/_openapi/responses.py:197]: Too broad

**Review by Goldziher — APPROVED** [review]

**Pull request merged**